genetic_algorithms/algorithms/simulatedannealing.py

```diff
@@ -0,0 +1,53 @@
+import math
+from genetic_algorithms.models.algorithm import Algorithm
+from genetic_algorithms.problems.base.state import State
+from genetic_algorithms.problems.base.model import Model
+from genetic_algorithms.helpers import History
+from typing import Tuple
+import random
+
+
+HISTORY_SIZE = 5
+
+
+class SimulatedAnnealing(Algorithm):
+    def __init__(self):
+        super(SimulatedAnnealing, self).__init__()
+        self.history = History[int](HISTORY_SIZE)
+        # how to set temperature ???
+        self.temperature = 100
+
+    def _simulated_annealing(self, model: Model, state: State) -> Tuple[State, int]:
+        best_state = state
+        best_state_cost = model.cost_for(state)
+        move_iteration_number = 0
+        for move in model.move_generator.available_moves(state):
+            new_state = move.make()
+            new_state_cost = model.cost_for(new_state)
+            if new_state_cost <= best_state_cost:
+                (best_state, best_state_cost) = (new_state, new_state_cost)
+            else:
+                # difference between new state cost and current state cost
+                diff = new_state_cost - best_state_cost
+                # calculate temperature for current iteration
+                self.temperature = self.temperature / float(move_iteration_number + 1)

```

Comment:
The cooling schedule is also a thing that could be injected.
In my opinion, you should extract the schedule into a separate class, e.g. `CoolingSchedule` that handles the temperature (both initial value and its changes) and inject it into the algorithm. In the basic version it could just have one method:
`getTemperatureForIteration(iteration: int) -> double`